src/js/server/schemaMiddleware.js

```diff
@@ -0,0 +1,138 @@
+    gradeNames: ["gpii.schema.handler"],
+    invokers: {
+        handleRequest: {
+            func: "{that}.sendResponse",
+            args: ["{that}.options.statusCode", "{that}.options.message"]
+        }
+    }
+});
+
+fluid.registerNamespace("gpii.schema.middleware");
+
+gpii.schema.middleware.rejectOrForward  = function (that, req, res, next) {
+    if (that.options.schemaDir && that.options.schemaKey) {
+        var toValidate = fluid.model.transformWithRules(req, that.options.rules.requestContentToValidate);
+        var results = that.validator.validate(that.options.schemaKey, toValidate);
+        if (results) {
+            var transformedResults = fluid.model.transformWithRules(results, that.options.rules.validationErrorsToResponse);
+
+            // Instantiate a handler that will take care of the rest of the request.
+            that.events.onInvalidRequest.fire(req, res, 400, transformedResults);
+        }
+        else {
+            next();
+        }
+    }
+    else {
+        // We choose to fail if we can't validate the body.  That way anything downstream is guaranteed to
+        // only receive valid content.
+        var message = "Your gpii.schema.middleware instance isn't correctly configured, so it can't validate anything.";
+        that.events.onInvalidRequest.fire(req, res, 500, { ok: false, error: message });
+    }
+};
+
+fluid.defaults("gpii.schema.middleware", {
+    gradeNames: ["gpii.express.middleware"],
+    responseSchemaKey: "message.json",
+    responseSchemaUrl: "http://terms.raisingthefloor.org/schema/message.json",
+    distributeOptions: [
+        {
+            source: "{that}.options.responseSchemaKey",
+            target: "{that gpii.schema.middleware.handler}.options.schemaKey"
+        },
+        {
+            source: "{that}.options.responseSchemaUrl",
+            target: "{that gpii.schema.middleware.handler}.options.schemaUrl"
+        }
+    ],
+    messages: {
+        error: "The JSON you have provided is not valid."
+    },
+    rules: {
+        validationErrorsToResponse: {
+            "": "",
+            "ok": {
+                literalValue: false
+            },
+            "message": {
+                literalValue: "{that}.options.messages.error"
+            }
+        },
+        requestContentToValidate: {
+            "": "body"
+        }
+    },
+    components: {
+        validator: {
+            type: "gpii.schema.validator.server",
+            options: {
+                schemaDir: "{gpii.schema.middleware}.options.schemaDir"
+            }
+        }
+    },
+    invokers: {
+        middleware: {
+            funcName: "gpii.schema.middleware.rejectOrForward",
+            args:     ["{that}", "{arguments}.0", "{arguments}.1", "{arguments}.2"]
+        }
+    },
+    // Approach adapted from `gpii.express.requestAware.router`.
+    events: {
+        "onInvalidRequest": null
+    },
+    dynamicComponents: {
+        requestHandler: {
```

Comment:
Dubious - why do we duplicate this material from the router? It shouldn't be necessary for something which acts a simple middleware to construct request components